src/jacobi.c

```diff
@@ -0,0 +1,219 @@
+ * @param size size of the matrix (matrix must be square)
+ * @param outputMatrix matrix containing every eigenvector of the input matrix (same size as the input matrix)
+ * @param iterations number of iterations of the algorithm
+ * @param cyclic whether to use the cyclic jacobi method or the classical version
+*/
+void jacobi(real_number* inputMatrix, vec_size size, real_number* outputMatrix, vec_size iterations, vec_size cyclic) {
+
+    vec_size sizes3d[3] = {size, size, size};
+    vec_size sweepSize = size * (size - 1) / 2; // A sweep is defined as a n * (n - 1) / 2 jacobi rotations
+    real_number p[size * size]; // Rotation matrix
+    real_number vBuffer[size * size];
+    real_number aBuffer[size * size];
+    vec_size maxRow, maxCol, indexToMinimize;
+    vec_size sizeSquared = size * size;
+
+    jacobiCreateIdentityMatrix(size, outputMatrix);
+
+    for (vec_size i = 0; i < iterations; ++i) {
+
+        if (cyclic) {
+            if (indexToMinimize == (sizeSquared - 2)) {
+                indexToMinimize = 0;
+            }
+
+            indexToMinimize = (indexToMinimize + 1) % (sizeSquared);
+            if (indexToMinimize / size == indexToMinimize % size) {
+                indexToMinimize = (indexToMinimize + 1) % (sizeSquared);
+            }
+
+            maxRow = indexToMinimize / size;
+            maxCol = indexToMinimize % size;
+        } else {
+            jacobiMaxIndex(inputMatrix, size, &maxRow, &maxCol);
+        }
+
+        if (i < 3 * sweepSize) {
+            // In the first three sweeps, we keep the rotation matrix if |a_pq| > epsilon
+            real_number epsilon_sweep = 0.20 * (jacobiComputeOffDiagonalSum(inputMatrix, size)) / ((real_number) (sizeSquared));
+            if (inputMatrix[coordToIndex(maxRow, maxCol, size)] <= epsilon_sweep) {
+                jacobiCreateRotationMatrix(inputMatrix, maxRow, maxCol, p, size);
+            }
+        } else {
+
+            real_number apq = real_abs(inputMatrix[coordToIndex(maxRow, maxCol, size)]);
+            real_number app = inputMatrix[coordToIndex(maxRow, maxRow, size)];
+            real_number aqq = inputMatrix[coordToIndex(maxCol, maxCol, size)];
+
+            // If |apq| << |app| and |apq << |aqq|, set the element to 0 and continue with other element
+            if (apq < pow(10, -(DIGITS_PRECISION + 2)) * app && apq < pow(10, -(DIGITS_PRECISION + 2) * aqq)) {
+                inputMatrix[coordToIndex(maxRow, maxCol, size)] = 0;
+                i--;
+                continue;
+            }
+
+            jacobiCreateRotationMatrix(inputMatrix, maxRow, maxCol, p, size);
+        }
+
+        jacobiMatrixMultiply(p, inputMatrix, sizes3d, aBuffer, 1);
+        jacobiMatrixMultiply(aBuffer, p, sizes3d, inputMatrix, 0);
+        jacobiMatrixMultiply(outputMatrix, p, sizes3d, vBuffer, 0);
+        memcpy(outputMatrix, vBuffer, sizeof(real_number) * sizeSquared);
+    }
+
+}
```

Comment:
No newline at the end of file ? 

Absolutely barbaric. If you're using vscode there's an option to automatically append newlines at the end, I think you should enable that ;)